src/main/java/com/charlesgadeken/entwined/EngineController.java

```diff
@@ -0,0 +1,146 @@
+        //            patternIndex++;
+        //        }
+        //        lx.engine.getChannel(channelIndex).goIndex(patternIndex);
+        // Trying....
+        lx.engine.mixer.setFocusedChannel(lx.engine.mixer.getChannel(channelIndex));
+        // ??
+    }
+
+    public void setChannelVisibility(int channelIndex, double visibility) {
+        lx.engine.mixer.getChannel(channelIndex).fader.setValue(visibility);
+    }
+
+    public void setActiveColorEffect(int effectIndex) {
+        if (activeEffectControllerIndex == effectIndex) {
+            return;
+        }
+        if (activeEffectControllerIndex != -1) {
+            TSEffectController effectController =
+                    effectControllers.get(activeEffectControllerIndex);
+            effectController.setEnabled(false);
+        }
+        activeEffectControllerIndex = effectIndex;
+        if (activeEffectControllerIndex != -1) {
+            TSEffectController effectController =
+                    effectControllers.get(activeEffectControllerIndex);
+            effectController.setEnabled(true);
+        }
+    }
+
+    public void setSpeed(double amount) {
+        speedEffect.speed.setValue(amount);
+    }
+
+    public void setSpin(double amount) {
+        spinEffect.spin.setValue(amount);
+    }
+
+    public void setBlur(double amount) {
+        blurEffect.level.setValue(amount);
+    }
+
+    public void setScramble(double amount) {
+        scrambleEffect.amount.setValue(amount);
+    }
+
+    public void setAutoplay(boolean autoplay) {
+        setAutoplay(autoplay, false);
+    }
+
+    void setAutoplay(boolean autoplay, boolean forceUpdate) {
+        if (autoplay != isAutoplaying || forceUpdate) {
+            isAutoplaying = autoplay;
+            automation.setPaused(!autoplay);
+
+            if (previousChannelIsOn == null) {
+                previousChannelIsOn = new boolean[lx.engine.mixer.getChannels().size()];
+                for (LXAbstractChannel channel : lx.engine.mixer.getChannels()) {
+                    previousChannelIsOn[channel.getIndex()] = channel.enabled.isOn();
+                }
+            }
+
+            for (LXAbstractChannel channel : lx.engine.mixer.getChannels()) {
+                boolean toEnable;
+                if (channel.getIndex() < baseChannelIndex) {
+                    toEnable = autoplay;
+                } else if (channel.getIndex() < baseChannelIndex + numChannels) {
+                    toEnable = !autoplay;
+                } else {
+                    toEnable = autoplay;
+                }
+
+                if (toEnable) {
+                    channel.enabled.setValue(previousChannelIsOn[channel.getIndex()]);
+                } else {
+                    previousChannelIsOn[channel.getIndex()] = channel.enabled.isOn();
+                    channel.enabled.setValue(false);
+                }
+            }
+
+            // @Slee lx.engine.getEffects() doesn't seem to exist any more:
```

Comment:
Those top-level global effects are now effects on the master channel.

`lx.engine.mixer.masterBus.effects` will get those for you

Reply:
Thank you! I couldn't find that for the life of me...